src/index/diskann/diskann.cc

```diff
@@ -384,40 +387,56 @@ DiskANNIndexNode<T>::Prepare(const Config& cfg) {
         return false;
     }
     if (num_nodes_to_cache > 0) {
-        std::vector<uint32_t> node_list;
         LOG_KNOWHERE_INFO_ << "Caching " << num_nodes_to_cache << " sample nodes around medoid(s).";
-        if (prep_conf.use_bfs_cache) {
-            auto gen_cache_expect = TryDiskANNCall<bool>([&]() -> bool {
-                pq_flash_index_->cache_bfs_levels(num_nodes_to_cache, node_list);
+        auto cached_nodes_file = diskann::get_cached_nodes_file(index_prefix_);
+        if (file_exists(cached_nodes_file)) {
+            LOG_KNOWHERE_INFO_ << "Reading cached nodes from file.";
+            auto gen_cache_from_file_expect = TryDiskANNCall<bool>([&]() -> bool {
+                pq_flash_index_->load_cache_list_from_file(cached_nodes_file, num_nodes_to_cache);
```

Comment:
During load, if file existed, we don't need to know how many cache to load. Just load all of them

Reply:
done